pkg/controller/ec2/transitgateway/setup.go

```diff
@@ -0,0 +1,145 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// SetupTransitGateway adds a controller that reconciles TransitGateway.
+func SetupTransitGateway(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(svcapitypes.TransitGatewayGroupKind)
+	opts := []option{
+		func(e *external) {
+			c := &custom{client: e.client, kube: e.kube}
+			e.postObserve = c.postObserve
+			e.postCreate = c.postCreate
+			e.preCreate = c.preCreate
+			e.isUpToDate = isUpToDate
+			e.lateInitialize = LateInitialize
+			e.filterList = filterList
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		For(&svcapitypes.TransitGateway{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.TransitGatewayGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.EC2API
+}
+
+func filterList(cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput) *svcsdk.DescribeTransitGatewaysOutput {
+	connectionIdentifier := aws.String(meta.GetExternalName(cr))
+	resp := &svcsdk.DescribeTransitGatewaysOutput{}
+	for _, TransitGateway := range obj.TransitGateways {
+		if aws.StringValue(TransitGateway.TransitGatewayId) == aws.StringValue(connectionIdentifier) {
+			resp.TransitGateways = append(resp.TransitGateways, TransitGateway)
+			break
+		}
+	}
+	return resp
+}
+
+func (e *custom) postObserve(_ context.Context, cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclients.StringValue(obj.TransitGateways[0].State) {
+	case string(svcapitypes.TransitGatewayState_available):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.TransitGatewayState_pending):
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.TransitGatewayState_modifying):
+		cr.SetConditions(xpv1.Unavailable())
+	case string(svcapitypes.TransitGatewayState_deleting):
+		cr.SetConditions(xpv1.Deleting())
+	case string(svcapitypes.TransitGatewayState_deleted):
+		// TransitGateway is in status deleted - and really removed after 6 hours in aws
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	return obs, nil
+}
+
+func isUpToDate(cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput) (bool, error) {
+	return true, nil
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.TransitGateway, obj *svcsdk.CreateTransitGatewayInput) error {
+	// set external name as tag on the transit gateway
```

Comment:
Let's add a [`tagger`](https://github.com/crossplane/provider-aws/blob/master/pkg/controller/ecr/repository/controller.go#L242) to do this during initialization.